FantasyUniverse/Services/Authentication/FirebaseAuthService.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  FirebaseAuthService.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 17.05.2022.
+//
+
+import Foundation
+import FirebaseAuth
+import Firebase
+
+class FirebaseAuthService {
+    
+    func createUser(email: String, password: String, completion: @escaping (_ success: Bool) -> Void) {
+        Auth.auth().createUser(withEmail: email, password: password) {(authResult, error) in
+            if let error = error as NSError? {
+                switch AuthErrorCode(rawValue: error.code) {
+                    case .operationNotAllowed:
+                        debugPrint("Error: The given sign-in provider is disabled for this Firebase project. Enable it in the Firebase console, under the sign-in method tab of the Auth section.")
+                    case .emailAlreadyInUse:
+                        debugPrint("Error: The email address is already in use by another account.")
+                    case .invalidEmail:
+                        debugPrint("Error: The email address is badly formatted.")
+                    case .weakPassword:
+                        debugPrint("Error: The password must be 6 characters long or more.")
+                    default:
+                        debugPrint("Error: \(error.localizedDescription)")
+                }
+                completion(false)
+            } else {
+                print(authResult?.user ?? "Firebase Sign up failed")
+                completion(true)
+            }
+        }
+    }
+    
+    func storeUserInfo(name: String, surname: String) {
+        
+        if userLoggedIn() {
+            let ref = Database.database().reference()
+            let userData = ["firstName": name,
+                            "lastName": surname]
+            ref.child("users").child(Auth.auth().currentUser!.uid).setValue(userData)
+        } else {
+            debugPrint("Error: Trying to store user info for signed out user")
+        }
+    }
+    
+    func signIn(email: String, pass: String, completionBlock: @escaping (_ success: Bool) -> Void) {
+        Auth.auth().signIn(withEmail: email, password: pass) { (result, error) in
```

Comment:
Isn't there any token returned or is it handled by Firebase internally?

Reply:
Yeah, we can retrieve token from result like result?.user.refreshToken or get idToken like firebase.auth().currentUser.getIdToken, but i will handle it later on